File: Niel/progamming/H24/in1000/innleveringer/innlevering 8/celle.py

class Celle:
    # Konstruktøren med attributter for status, naboer og antall levende naboer.
    def __init__(self):
        self._status = "doed"
        self._naboer = []
        self._ant_levende_naboer = 0

    # setter statusen til cellen til "doed".
    def sett_doed(self):
        self._status = "doed"

    # setter statusen til cellen til "levende".
    def sett_levende(self):
        self._status = "levende"

    # returnerer True hvis cellens status er "levende", ellers False.
    def er_levende(self):
        if self._status == "levende":
            return True
        else:
            return False

    # returnerer "O" hvis cellen er "levende", ellers returnerer ".".
    def hent_status_tegn(self):
        if self._status == "levende":
            return "O"
        else:
            return "."

    # legger til en nabo til cellens nabo-liste.
    def legg_til_nabo(self, nabo):
        self._naboer.append(nabo)

    # teller og oppdaterer antallet levende naboer rundt cellen.
    def tell_levende_naboer(self):
        self._ant_levende_naboer = 0
        for _ in self._naboer:
            if _._status == "levende":
                self._ant_levende_naboer += 1

    # oppdaterer cellens status basert oppgaveteksten
    def oppdater_status(self):
        if self.er_levende():
            if self._ant_levende_naboer < 2 or self._ant_levende_naboer > 3:
                self.sett_doed()
        else:
            if self._ant_levende_naboer == 3:
                self.sett_levende()

File: Niel/progamming/V25/in1010/innleveringer/oblig 1/celle.java
class Celle {

    boolean levende; // Variabel for om cellen er levende
    Celle[] naboer; // Array for naboceller
    int antNaboer; // Variabel til å telle antall naboer
    int antLevendeNaboer; // Variabel til å telle antall levende naboer.

    public Celle(){
        levende = false; // Setter cellen som død ved initiering.
        antNaboer = 0; // Initierer antall naboer til 0.
        antLevendeNaboer = 0; // Initierer antall levende naboer til 0.
        naboer = new Celle[8]; // Skaper en array for å lagre inntil 8 naboer.
    }

    // Setter statusen til cellen som død.
    public void settDød(){
        levende = false;
    }

    // Setter statusen til cellen som levende.
    public void settLevende(){
        levende = true;
    }

    // Returnerer true hvis cellen er levende, ellers false.
    public boolean erLevende(){
        if (levende == true){
            return true;
        } else {
            return false;
        }
    }

    // Returnerer 'O' hvis cellen er levende og '.' hvis den er død.
    public char hentStatusTegn(){
        if (levende == true){
            return 'O';
        } else {
            return '.';
        }
    }

    // Legger til en nabocelle til naboer-arrayen hvis det er plass.
    public void leggTilNabo(Celle nabo) {
        if (naboer.length > antNaboer){
            naboer[antNaboer] = nabo;
            antNaboer++;
        } else {
            System.out.println("Kan ikke ha flere naboer");
        }
    }

    // Teller antall levende naboer ved å iterere gjennom naboer-arrayen.
    public void tellLevendeNaboer(){
        antLevendeNaboer = 0;

        for (int i = 0; i < antNaboer; i++){
            if (naboer[i].erLevende()){
                antLevendeNaboer++;
            }
        }
    }

    // Oppdaterer statusen til cellen; blir levende hvis den har nøyaktig 3 levende naboer, ellers blir den død.
    public void oppdaterStatus() {

        if (this.erLevende()){
            if (antLevendeNaboer < 2){
                settDød();
            } else if (antLevendeNaboer > 3) {
                settDød();
            }
        } else { // om  cellen er død
            if (antLevendeNaboer == 3) {
                settLevende();
        }

    }
    }
}
